README.md

```diff
@@ -1 +1,1380 @@
+
+## <a id="summary"></a>Summary
+
+[Back to the Top](#top)
+
+This section is a briefed overview of different scripting aspects in the three products.
+
+The [References](#references) section contains collection of links to the official scripting documentation. The links are organized by product and by area of concern.
+
+* ### <a id="summary-application-and-environment"></a>Application and Environment
+
+    Scripting application is augmenting or extending native functionality of a product. The extension points could be events associated with an object or a procedure; for example, an update event on a managed object in IDM. Or, the scripts could represent an essential part of a procedure, necessary for its successful completion; for example, scripts employed in AM authentication flows.
+
+    Scripts' environment could be described as access to methods and data. This may depend on the context the script is running in: the product and the particular procedure the script is extending. Even within the same product, the context may vary as the product components implement different functionality and expose different APIs.
+
+    ### <a id="summary-application-and-environment-am"></a> AM
+
+    Scripting application in AM could be summarized into the following categories:
+
+    * Authentication
+
+        * Modules and Chains
+            * Client-side
+            * Server-side
+        * Trees
+            * Client-side
+            * Server-side
+
+    * Authorization
+
+        * Scripting Policy Condition
+        * Access Token Modification
+
+    * Federation
+
+        * OIDC Claims Handling
+
+    #### <a id="am-scripting-client-side"></a>AM > Client-Side Scripts
+
+    AM may be used for authentication in the front channel. This can be assisted with custom client-side scripts written in JavaScript and executed in the user agent.
+
+    The use case for a client-side script is collecting information about the user agent's properties and its environment: [Geolocation](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/geolocation), IP, and whatever else that could be collected with a custom script running in a browser. Thus, the script needs to be written in JavaScript compatible with the browser, has access to the browser properties, and can make requests to external network resources _from the browser_. It could be used, for example, for detecting the client's IP.
+
+    The data collected by a client-side script can be submitted to the server side and become available for the server components involved in the same authentication procedure.
+
+    Scriptable access to the browser environment seems to be a unique feature of authentication procedures in AM, in comparison to the other script applications in the three products.
```

Comment:
"seems to be a unique"? I think it's fair to be definitive and just say that it is unique.

Reply:
Yes, I guess I was talking to you at this point. If I can say that definitively, I'll remove the "seems".